ui/src/criteria/valueData.tsx

```diff
@@ -219,104 +234,138 @@ export function ValueDataEdit(props: ValueDataEditProps) {
-                {props.valueConfigs?.map((c) =>
-                  hintDataState.data?.hintData?.find(
-                    (hint) => hint.attribute === c.attribute
-                  ) ? (
-                    <MenuItem key={c.attribute} value={c.attribute}>
-                      {c.title}
-                    </MenuItem>
-                  ) : null
-                )}
-              </Select>
-            </FormControl>
-          ) : null}
-          {selectedConfigs.map((c, i) => {
-            let component: ReactNode = null;
-            if (c.hintData.enumHintOptions) {
-              component = (
-                <HintDataSelect
-                  key={c.valueConfig.attribute}
-                  hintData={c.hintData}
-                  selected={c.valueData.selected}
-                  onSelect={(sel) => onValueSelect(sel, c.valueData)}
-                />
-              );
-            }
-            if (c.hintData.integerHint) {
-              component = (
-                <RangeSlider
-                  key={c.valueConfig.attribute}
-                  index={0}
-                  minBound={minBound(c.attribute, c.hintData)}
-                  maxBound={maxBound(c.attribute, c.hintData)}
-                  range={c.valueData.range}
-                  unit={c.valueConfig.unit}
-                  onUpdate={(range, index, min, max) =>
-                    onUpdateRange(range, index, min, max, c.valueData)
+                <Checkbox
+                  size="small"
+                  fontSize="inherit"
+                  checked={!!props.valueData}
+                  checkedIcon={<FilterListIcon />}
+                  uncheckedIcon={<FilterListOffIcon />}
+                  onChange={() =>
+                    props.update(props.valueData ? undefined : [ANY_VALUE_DATA])
                   }
                 />
-              );
-            }
-            if (!component || props.singleValue) {
-              return component;
-            }
-
-            return (
-              <GridLayout key={c.valueData.attribute} rows height="auto">
-                {i !== 0 ? (
-                  <Divider variant="middle">
-                    <Chip label="AND" />
-                  </Divider>
-                ) : null}
-                <GridLayout cols rowAlign="middle" spacing={3} height="auto">
-                  {!!c.valueConfig.title ? (
-                    <Typography variant="body1">
-                      {c.valueConfig.title}
-                    </Typography>
-                  ) : null}
-                  {component}
-                </GridLayout>
-              </GridLayout>
-            );
-          })}
+              </GridBox>
+            ) : null}
+          </Loading>
           <GridBox />
         </GridLayout>
       ) : null}
-    </Loading>
+      <Loading status={hintDataState} noProgress={!!props.title}>
+        {hasHints && (!props.title || isValid(props.valueData)) ? (
+          <GridLayout rows height="auto">
```

Comment:
Other than forcing a row based layout here to better fit in the current, narrow UI, the code for the value editing controls is unchanged.